UIKitDZ/UIKitDZ/Controller/ViewController.swift

```diff
@@ -0,0 +1,159 @@
+        // Подключаю делегат и датасорс для работы с таблицой
+        tableView.dataSource = self
+        tableView.delegate = self
+        view.addSubview(tableView)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+
+        // MARK: Констрейнты для полного покрытие экрана тейблвью
+
+        NSLayoutConstraint.activate([
+            tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+
+        ])
+    }
+
+    // MARK: Функция для создание хедера
+
+    fileprivate func createHeaderButton() {
+        /// Логотип приложения с кастомным шрифтом
+        let logoLabel = UILabel()
+        logoLabel.text = "RMLink"
+        logoLabel.font = UIFont(name: "Dancing Script", size: 22)
+        logoLabel.sizeToFit()
+
+        let leftBarButton = UIBarButtonItem(customView: logoLabel)
+
+        /// Кнопка сообщение верху
+        let imageButton = UIButton()
+        let image = UIImage(named: "message")
+        imageButton.setImage(image, for: .normal)
+        imageButton.tintColor = .black
+        imageButton.widthAnchor.constraint(equalToConstant: 18).isActive = true
+        imageButton.heightAnchor.constraint(equalToConstant: 18).isActive = true
+        navigationItem.leftBarButtonItem = leftBarButton
+        navigationItem.rightBarButtonItem = UIBarButtonItem(customView: imageButton)
+    }
+
+    @objc func imageButtonTapped() {
+        print("Скоро")
+    }
+}
+
+// MARK: Расширение ViewController
+
+extension ViewController {
+    /// Типы постов
+    enum PostTypes {
+        case stories
+        case firstPost
+        case recommendationPost
+        case otherPosts
+    }
+}
+
+// MARK: Расширение для реализации UITableViewDataSource
+
+extension ViewController: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        postTypes.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch postTypes[section] {
+        case .stories, .firstPost, .recommendationPost:
+            return 1
+        case .otherPosts:
+            return source.posts.count
+        }
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch postTypes[indexPath.section] {
+        case .stories:
+            guard let cell = tableView.dequeueReusableCell(
+                withIdentifier: "StoryViewCell",
+                for: indexPath
+            ) as? StoryViewCell
+            else { return UITableViewCell() }
+            let post = source.story
+            cell.setupCell(posts: post)
```

Comment:
```suggestion
            cell.setup(with: post)
```
Тут и так понятно что ты настраиваешь ячейку